lua/weapons/cfc_stinger_launcher.lua

```diff
@@ -0,0 +1,464 @@
+
+    for _, vehicle in ipairs( ents.FindByClass( "npc_helicopter" ) ) do
+        table.insert( foundVehicles, vehicle )
+    end
+
+    for _, vehicle in ipairs( ents.FindByClass( "npc_combinegunship" ) ) do
+        table.insert( foundVehicles, vehicle )
+    end
+
+    for _, vehicle in ipairs( ents.FindByClass( "npc_combinedropship" ) ) do
+        table.insert( foundVehicles, vehicle )
+    end
+
+    for _, vehicle in ipairs( ents.FindByClass( "prop_vehicle_*" ) ) do
+        local vechiclesDriver = vehicle:GetDriver()
+        if IsValid( vechiclesDriver ) then
+            local parent = vehicle:GetParent()
+            if parent:IsVehicle() and parent:GetDriver() == vehiclesDriver and not addedAlready[parent] then -- glide/simfphys
+                table.insert( foundVehicles, parent )
+                addedAlready[parent] = true
+                addedAlready[vehicle] = true
+            else
+                table.insert( foundVehicles, vehicle )
+            end
+        end
+    end
+    return foundVehicles
+end
+
+function SWEP:Think()
+    if CLIENT then return end
+
+    self.nextSortTargets = self.nextSortTargets or 0
+    self.findTime = self.findTime or 0
+    self.nextFind = self.nextFind or 0
+
+    local curtime = CurTime()
+    local owner = self:GetOwner()
+    local findTime = self.findTime
+    local lockOnTime = stingerLockTime:GetFloat()
+
+    if findTime + lockOnTime < curtime and IsValid( self:GetClosestEnt() ) then
+        self.Locked = true
+    else
+        self.Locked = false
+    end
+
+    if self.Locked ~= self:GetIsLocked() then
+        self:SetIsLocked( self.Locked )
+
+        if self.Locked then
+            self.LockSND = CreateSound( owner, "weapons/cfc_stinger/radar_lock.wav" )
+            self.LockSND:PlayEx( 0.5, 100 )
+
+            if self.TrackSND then
+                self.TrackSND:Stop()
+                self.TrackSND = nil
+            end
+        else
+            if self.LockSND then
+                self.LockSND:Stop()
+                self.LockSND = nil
+            end
+        end
+    end
+
+    if self.nextFind < curtime then
+        self.nextFind = curtime + 3
+        self.foundVehicles = self:GetPotentialTargets()
+    end
+
+    if self:Clip1() <= 0 then
+        self:SetClosestEnt( nil )
+        if self.TrackSND then
+            self.TrackSND:Stop()
+            self.TrackSND = nil
+        end
+
+    elseif self.nextSortTargets < curtime then
+        self.nextSortTargets = curtime + 0.25
+        self.foundVehicles = self.foundVehicles or {}
+
+        local AimForward = owner:GetAimVector()
+        local startpos = owner:GetShootPos()
+
+        local maxDist = GetGlobalInt( "cfc_stinger_maxrange" )
+        local lockOnAng = stingerLockAngle:GetInt()
+
+        local vehicles = {}
+        local closestEnt = NULL
+        local closestDist = math.huge
+        local smallestAng = math.huge
+
+        for index, vehicle in ipairs( self.foundVehicles ) do
```

Comment:
Removing indices will break when using `ipairs()`
```suggestion
        for i = #self.foundVehicles, 1, -1 do
            local vehicle = self.foundVehicles[i]

```
Then update `index` -> `i` and localize `self.foundVehicles`